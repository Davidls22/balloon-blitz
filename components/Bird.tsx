import React from 'react';
import { View, Image } from 'react-native';

type Props = {
  left: number;
  bottom: number;
  isUp: boolean;
};

export default function Bird({ left, bottom, isUp }: Props) {
  const size = 40;

  return (
    <View
      style={{
        position: 'absolute',
        width: size,
        height: size,
        left,
        bottom,
      }}
    >
      <Image
        source={
          isUp
            ? require('../assets/images/BirdUp.png')
            : require('../assets/images/BirdDown.png')
        }
        style={{ width: '100%', height: '100%' }}
        resizeMode="contain"
      />
    </View>
  );
}