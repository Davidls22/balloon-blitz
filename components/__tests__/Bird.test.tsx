import React from 'react';
import renderer from 'react-test-renderer';
import Bird from '../Bird';

test('renders bird up', () => {
  const tree = renderer.create(<Bird left={50} bottom={100} isUp />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders bird down', () => {
  const tree = renderer.create(<Bird left={50} bottom={100} isUp={false} />).toJSON();
  expect(tree).toMatchSnapshot();
});