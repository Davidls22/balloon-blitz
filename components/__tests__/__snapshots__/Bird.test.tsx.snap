// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders bird down 1`] = `
<View
  style={
    {
      "bottom": 100,
      "height": 40,
      "left": 50,
      "position": "absolute",
      "width": 40,
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      {
        "testUri": "../../../assets/images/BirdDown.png",
      }
    }
    style={
      {
        "height": "100%",
        "width": "100%",
      }
    }
  />
</View>
`;

exports[`renders bird up 1`] = `
<View
  style={
    {
      "bottom": 100,
      "height": 40,
      "left": 50,
      "position": "absolute",
      "width": 40,
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      {
        "testUri": "../../../assets/images/BirdUp.png",
      }
    }
    style={
      {
        "height": "100%",
        "width": "100%",
      }
    }
  />
</View>
`;
